### PR TITLE
Replace hard coded project items

### LINKED ISSUE #133: Replace hardcoded sample project items count in SelectedProject.js
### Motivation

In */src/Utilities/SelectedProject.js* there is a hardcoded SAMPLE_PROJECT_ITEM_COUNT constant which is used in the wizard. 

This wizard checks if the Delivery API has already accessible all items in the user's Dancing Goat project and if the project is fully accessible. If the count of all items in the project is more or equals to items from constant then we consider the project to be ready.

However, if the user's sample Dancing Goat project changes (we add/remove new content) this logic might not work.  Under the hood, we clone the Dancing Goat project from 'one-source-of-truth' project.

### Proposed solution

The goal of this issue is to implement logic which will get a count of items in from this source (-of-truth) project (projectId: 975bf280-fd91-488c-994c-2f04416e5ee3) and store it to an existing (non)constant which will be checked the same way as it is right now.

Note: It might be a good idea to adjust if condition in *Configuration.js*, too.
Something like ... if (response.items.length > SAMPLE_PROJECT_ITEM_COUNT) in *waitUntilProjectAccessible* function.

### PR DESCRIPTION
**Pull request by sheidaee**
### Motivation

Fixes #133 by removing the constant in SelectedProject.js and adding an API call when the waitUntilProjectAccessible function is invoked.

### Checklist

- [ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [ ] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

Test creating a new project based on the sample project and test the new API call returns the correct values. Recommend maybe adding a few items to the project too to test the updated if condition.

### CODE REVIEW

File: src/Utilities/SelectedProject.js [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
+    .depthParameter(0)
+    .getObservable();  
+}
 
 export {
   selectedProjectCookieName,
   projectConfigurationPath,
-  defaultProjectId,
-  SAMPLE_PROJECT_ITEM_COUNT
+  defaultProjectId,
```
Comment (makma) [inline src/Utilities/SelectedProject.js]:
```suggestion
  defaultProjectId,
```

File: src/Pages/Admin/Configuration.js [hunk 2]
```diff
@@ -108,15 +108,22 @@ class Configuration extends Component {
...
-          if (response.items.length === SAMPLE_PROJECT_ITEM_COUNT) {
-            this.setState({
-              preparingProject: false
-            });
-            this.redirectToHome(newProjectId);
-          } else {
-            this.waitUntilProjectAccessible(newProjectId);
-          }
+        .subscribe(response => {          
+          const sampleProjectClient = getSampleProjectItems()
```
Comment (makma) [inline src/Pages/Admin/Configuration.js]:
```suggestion
          const sampleProjectListingResponse= getSampleProjectItems()          
```

File: src/Pages/Admin/Configuration.js [hunk 3]
```diff
@@ -108,15 +108,22 @@ class Configuration extends Component {
...
-              preparingProject: false
-            });
-            this.redirectToHome(newProjectId);
-          } else {
-            this.waitUntilProjectAccessible(newProjectId);
-          }
+        .subscribe(response => {          
+          const sampleProjectClient = getSampleProjectItems()          
+            .subscribe(
+            (sampleProjectClientResult)=> {
```
Comment (makma) [inline src/Pages/Admin/Configuration.js]:
```suggestion
            (sampleProjectClientResult) => {                                          
```

File: src/Pages/Admin/Configuration.js [hunk 4]
```diff
@@ -108,15 +108,22 @@ class Configuration extends Component {
...
-            });
-            this.redirectToHome(newProjectId);
-          } else {
-            this.waitUntilProjectAccessible(newProjectId);
-          }
+        .subscribe(response => {          
+          const sampleProjectClient = getSampleProjectItems()          
+            .subscribe(
+            (sampleProjectClientResult)=> {                                          
+              if ( response.items.length >= sampleProjectClientResult.items.length) {
```
Comment (makma) [inline src/Pages/Admin/Configuration.js]:
```suggestion
              if (response.items.length >= sampleProjectClientResult.items.length) {                            
```

File: src/Pages/Admin/Configuration.js [hunk 5]
```diff
@@ -108,15 +108,22 @@ class Configuration extends Component {
...
-          }
+        .subscribe(response => {          
+          const sampleProjectClient = getSampleProjectItems()          
+            .subscribe(
+            (sampleProjectClientResult)=> {                                          
+              if ( response.items.length >= sampleProjectClientResult.items.length) {                            
+                this.setState({
+                  preparingProject: false
+                });              
+                sampleProjectClient.unsubscribe();
```
Comment (makma) [inline src/Pages/Admin/Configuration.js]:
```suggestion
                sampleProjectListingResponse.unsubscribe();
```

File: src/Pages/Admin/Configuration.js [hunk 6]
```diff
@@ -108,15 +108,22 @@ class Configuration extends Component {
...
+                this.redirectToHome(newProjectId);                        
+              } else {
+                sampleProjectClient.unsubscribe();
```
Comment (makma) [inline src/Pages/Admin/Configuration.js]:
```suggestion
                sampleProjectListingResponse.unsubscribe();
```

**Comment by sheidaee** [comment]
Hello @makma 
My pleasure. I have done the changes. as for `package-lock.json` file, is it OK if I just remove it from the files changes section?

**Comment by makma** [comment]
Hi @sheidaee, thank you for accepted changes, code looks good to me, however, there is still `package-lock.json` in the changes files.

**Comment by makma** [comment]
Hello @sheidaee, I appreciate your effort, nevertheless,  you have now deleted `package-lock.json` - _Delete file_ button in GitHub UI really deletes the file (and not changes only) - I believe this is the reason why the file is deleted.

**Comment by makma** [comment]
Looks good now, going to test it and let you know. Thank you @sheidaee

**Comment by makma** [comment]
It works perfectly now. Thank you for your contribution @sheidaee!

**Pull request merged**